TypeSystem/flow-analysis/reachability_switch_A08_t04.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion switch statement: If `N` is a switch statement of the form
+/// `switch (E) {alternatives}` then:
+/// - Let `before(E) = before(N)`.
+/// - For each `C` in `alternatives` with statement body `S`:
+///   - If `C` is labelled let
+///     `before(S) = conservativeJoin(after(E), assignedIn(N), capturedIn(N))`
+///     otherwise let `before(S) = after(E)`.
+///  - If the cases are exhaustive, then let `after(N) = break(N)` otherwise let
+///    `after(N) = join(after(E), break(N))`.
+///
+/// @description Checks that if a variable is initialized in a `when` part of a
+/// `case` statement and the switch statement is labeled then this variable is
+/// not definitely assigned.
+/// @author sgrekhov22@gmail.com
+
+void test1() {
+  int i;
+  switch (42) {
+    label: case _ when (i = 42) > 0:
+  }
+  i; // Not definitely assigned
+//^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+void test2() {
+  int i;
+  switch (42) {
+    label:
+    case String _:
+      break;
+    case _ when (i = 42) > 0:
+  }
+  i;
```

Comment:
If we remove the label and assume that the flow analysis does not know that the intersection between `String` and `int` is the empty type (`Never`) then `i` should still be not-definitely assigned at this point. In other words, it's not about the label (unless flow analysis knows that the labeled case is unreachable). Is there a "dead code" warning?

Otherwise I think it isn't really justified to say that we're testing anything about labels in this test case.

Reply:
Perhaps test some unrelated types where the flow analysis cannot know that the intersection is empty (`class A {}` and `class B {}` could have a shared subtype)? That seems to be a more readable test case.

Reply:
>  Is there a "dead code" warning?

No, there isn't. Removed `test2()` at all. `test1()` fails without the label (no expected error) and succeeds with it (there is an expected error).

Reply:
OK, sounds good!